src/function/aggregate/nested/list.cpp

```diff
@@ -474,6 +484,87 @@ static void BuildListVector(ReadDataFromSegment &read_data_from_segment, LinkedL
 	linked_list->last_segment = nullptr;
 }
 
+template <class T>
+static void CopyDataFromPrimitiveSegment(CopyDataFromSegment &, ListSegment *source, ListSegment *&target,
```

Comment:
Maybe you should return `target` instead of setting it as a side effect? And `source` is required and `const` so maybe that argument should have type `const ListSegment &`?